plugins/glustershd/actor.go

```diff
@@ -19,69 +21,148 @@ const (
 
 type shdActor struct{}
 
-func (actor *shdActor) Do(v *volume.Volinfo, key string, value string, logger log.FieldLogger) error {
-	if key != selfHealKey {
+func (actor *shdActor) Do(v *volume.Volinfo, key string, value string, volOp xlator.VolumeOpType, logger log.FieldLogger) error {
+	voltype := v.Type.String()
+
+	if voltype != "Replicate" && voltype != "Disperse" {
 		return nil
 	}
+
 	glustershDaemon, err := newGlustershd()
 	if err != nil {
 		return err
 	}
-
-	switch value {
-	case "on":
-		if isHealEnabled(v) {
-			err = daemon.Start(glustershDaemon, true, logger)
-			if err != nil && err != gderrors.ErrProcessAlreadyRunning {
+	switch volOp {
+	case xlator.VolumeStart:
+		for k, v := range v.Options {
```

Comment:
I didn't catch this earlier. You don't have to loop over all options here as it's a map. Specifically look for `replicate.self-heal-daemon` or `afr.self-heal-daemon`

Reply:
(this is not only for the self-heal daemon).This is intended to leave the action to registered actors. whether they need to start or stopped based on the default options.

Reply:
this is only for glustershd right? other actors will look for there keys to stop and start the daemons.

Reply:
@aravindavk  yes, its actors work to start/stop daemons. I think I need to edit my previous comment.
and update the code as per @prashanthpai comment